src/ui/handler/touch_zoom_rotate.js

```diff
@@ -147,34 +157,67 @@ class TouchZoomRotateHandler {
             }
 
             if (this._gestureIntent) {
+                this._map.fire(`${this._gestureIntent}start`, { originalEvent: e });
+                this._map.fire('movestart', { originalEvent: e });
                 this._startVec = vec;
-                this._startScale = map.transform.scale;
-                this._startBearing = map.transform.bearing;
+                this._map._startAnimation(this._onTouchFrame, this._onTouchFinished);
             }
+        }
 
-        } else {
-            const param: Object = { duration: 0, around: map.unproject(p) };
+        this._lastTouchEvent = e;
 
-            if (this._gestureIntent === 'rotate') {
-                param.bearing = this._startBearing + bearing;
-            }
-            if (this._gestureIntent === 'zoom' || this._gestureIntent === 'rotate') {
-                param.zoom = map.transform.scaleZoom(this._startScale * scale);
-            }
+        e.preventDefault();
+    }
```

Comment:
This needs a `this._map._update()` like in #6068, right?

Reply:
Yep 👍